frontend/src/libs/components/router/router.tsx

```diff
@@ -1,40 +1,50 @@
 import { Route } from 'react-router-dom';
 
 import { AppRoute } from '~/libs/enums/enums.js';
+import { useAppDispatch, useEffect } from '~/libs/hooks/hooks.js';
 import { UserGroupKey } from '~/packages/users/libs/enums/enums.js';
 import { Auth } from '~/pages/auth/auth.js';
 import { Dashboard } from '~/pages/dashboard/dashboard.js';
 import { NotFound } from '~/pages/not-found/not-found.js';
 import { WelcomePage } from '~/pages/welcome/welcome.js';
+import { actions as authActions } from '~/slices/auth/auth.js';
 
 import { App } from '../app/app.js';
 import { ProtectedRoute } from '../components.js';
 import { PageLayout } from '../page-layout/page-layout.js';
 import { RouterProvider } from '../router-provider/router-provider.js';
 
-const Router = (): JSX.Element => (
-  <RouterProvider>
-    <Route path={AppRoute.ROOT} element={<App />}>
-      <Route path={AppRoute.WELCOME} element={<WelcomePage />} />
-      <Route path={AppRoute.SIGN_IN} element={<Auth />} />
-      <Route path={AppRoute.SIGN_UP} element={<Auth />} />
-    </Route>
-    <Route
-      path={AppRoute.ROOT}
-      element={<ProtectedRoute allowedUserGroup={UserGroupKey.BUSINESS} />}
-    >
+const Router = (): JSX.Element => {
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
```

Comment:
It looks like you're applying this strategy to any possible session, right?
What if the current application user does not have account yet and wants to use the application anonymously?

I believe, we should apply this only if there is a jwt key in the localstorage. You should add this check here, and maybe you would consider to wrap it into a local hook (by local i mean it should rest in router/libs/hooks), just to make it look more concise.

Another option might be to apply this only for "protected" routes, however it looks like we don't have frontend protected functionality in public yet. Do you know, who is responsive for that?

Reply:
Good point, thank you